src/server/authentication.js

```diff
@@ -0,0 +1,114 @@
+const argon2 = require('argon2');
+const validator = require('validator');
+const jwt = require('jsonwebtoken');
+
+module.exports = psql => ({
+
+  /*
+      first checks if username/email already exist
+      returns true if successful
+     */
+  register: async (user) => {
+    const { username } = user;
+    const { email } = user;
+    const { password } = user;
+    let checkUsers;
+    let hash;
+
+    if (!validator.isEmail(email)) {
+      return false;
```

Comment:
elsewhere, on failure, a string is returned. It would be fair to let the user know what to do differently
```suggestion
      return 'the email address used is invalid';
```